20181209-ArmandXiao.java

```diff
@@ -0,0 +1,23 @@
+//part (a)
+public String decodeString(ArrayList<StringPart> parts){
+    String temp = "";
+    int start = 0;
+    int length = 0;
+    for( int i = 0, i< parts.size(), i++){
+      start = parts.get(i).getStart();
+      if( start != -1){
```

Comment:
start: the starting position of the substring in a master string, it always greater than or equals 0.